packages/fluentui/react-northstar/src/components/Dialog/Dialog.tsx

```diff
@@ -117,304 +132,334 @@ export const dialogSlotClassNames: DialogSlotClassNames = {
-    confirmButton: customPropTypes.itemShorthand,
-    defaultOpen: PropTypes.bool,
-    header: customPropTypes.itemShorthand,
-    onCancel: PropTypes.func,
-    onConfirm: PropTypes.func,
-    onOpen: PropTypes.func,
-    open: PropTypes.bool,
-    overlay: customPropTypes.itemShorthand,
-    trapFocus: PropTypes.oneOfType([PropTypes.bool, PropTypes.object]),
-    trigger: PropTypes.any,
-  };
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  });
 
-  static defaultProps = {
-    accessibility: dialogBehavior as Accessibility,
-    actions: {},
-    backdrop: true,
-    closeOnOutsideClick: true,
-    overlay: {},
-    footer: {},
-    trapFocus: true,
-  };
+  const [open, setOpen] = useAutoControlled({
+    defaultValue: props.defaultOpen,
+    value: props.open,
+    initialValue: false,
+  });
 
-  static autoControlledProps = ['open'];
-  static Footer = DialogFooter;
+  const lockBodyScroll = React.useCallback(() => {
+    const openDialogs = (+context.target.body.getAttribute(dialogsCounterAttribute) || 0) + 1;
+    context.target.body.setAttribute(dialogsCounterAttribute, `${openDialogs}`);
 
-  actionHandlers = {
-    closeAndFocusTrigger: e => {
-      this.handleDialogCancel(e);
-      e.stopPropagation();
+    // Avoid to block scroll in nested dialogs
+    if (openDialogs === 1) {
+      disableBodyScroll(context.target.body);
+    }
+  }, [context.target.body]);
 
-      _.invoke(this.triggerRef, 'current.focus');
-    },
-    close: e => this.handleDialogCancel(e),
-  };
-  contentRef = React.createRef<HTMLElement>() as React.MutableRefObject<HTMLElement>;
-  overlayRef = React.createRef<HTMLElement>() as React.MutableRefObject<HTMLElement>;
-  triggerRef = React.createRef<HTMLElement>();
-
-  getInitialAutoControlledState(): DialogState {
-    return {
-      open: false,
-    };
-  }
-
-  static getAutoControlledStateFromProps(props: DialogProps, state: DialogState): Partial<DialogState> {
-    return {
-      contentId: getOrGenerateIdFromShorthand('dialog-content-', props.content, state.contentId),
-      headerId: getOrGenerateIdFromShorthand('dialog-header-', props.header, state.headerId),
-    };
-  }
-
-  handleDialogCancel = (e: Event | React.SyntheticEvent) => {
-    _.invoke(this.props, 'onCancel', e, { ...this.props, open: false });
-    this.setState({ open: false });
+  const unlockBodyScroll = React.useCallback(() => {
+    const openDialogs = (+context.target.body.getAttribute(dialogsCounterAttribute) || 0) - 1;
+    context.target.body.setAttribute(dialogsCounterAttribute, `${openDialogs}`);
+
+    // Only enables scroll if all dialogs are closed
+    if (openDialogs === 0) {
+      enableBodyScroll(context.target.body);
+      context.target.body.removeAttribute(dialogsCounterAttribute);
+    }
+  }, [context.target.body]);
+
+  const [openedFirst, setOpenedFirst] = React.useState(props.open || props.defaultOpen);
```

Comment:
I don't understand this and it should not be required

Reply:
If not set, the first render open will be `false`, and then will trigger `unlockBodyScroll` and it will count -1 dialog so breaks the way it supposed to work. Before it worked because we were comparing with `prevState`

```typescript
if(prevState.open && !state.open){
    unlockBodyScroll();
}
```

Reply:
Don't use `else` there

```
React.useEffect(() => {
  if (open) lock()

  return () => {
    if (open) unlock()
  }
})
```

Reply:
You are right, done.